File: galaxy-web-app/galaxy-oj/src/main/java/io/charlie/web/oj/modular/data/reports/entity/TaskReports.java
package io.charlie.web.oj.modular.data.reports.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import io.charlie.galaxy.pojo.CommonEntity;

import java.math.BigDecimal;
import java.io.Serial;
import java.util.List;

import io.charlie.web.oj.modular.data.problem.entity.DataProblem;
import io.charlie.web.oj.modular.data.set.entity.DataSet;
import io.charlie.web.oj.modular.data.similarity.dto.CloneLevel;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.dromara.core.trans.anno.Trans;
import org.dromara.core.trans.constant.TransType;

/**
 * @author Charlie Zhang
 * @version v1.0
 * @date 2025-09-21
 * @description 报告库表
 */
@EqualsAndHashCode(callSuper = true)
@Data
@TableName(value = "task_reports", autoResultMap = true)
@Schema(name = "TaskReports", description = "报告库表")
public class TaskReports extends CommonEntity {
    @Serial
    @TableField(exist = false)
    private static final long serialVersionUID = 1L;

    @Schema(description = "主键")
    private String id;

    @Schema(description = "报告类型")
    @Trans(type = TransType.DICTIONARY, key = "REPORT_TYPE", ref = "reportTypeName")
    private Integer reportType;

    @Schema(description = "报告类型名称")
    @TableField(exist = false)
    private String reportTypeName;

    @Schema(description = "任务ID")
    private String taskId;

    @Schema(description = "模块类型")
    @Trans(type = TransType.DICTIONARY, key = "CODE_SUBMIT_MODULE")
    private String moduleType;
    @Schema(description = "模块ID")
    private String moduleId;

    @Schema(description = "题目ID")
    @Trans(type = TransType.SIMPLE, target = DataProblem.class, fields = "title", ref = "problemIdName")
    private String problemId;

    @Schema(description = "题目名称")
    @TableField(exist = false)
    private String problemIdName;

    @Schema(description = "样例数量")
    private Integer sampleCount;

    @Schema(description = "相似组数量")
    private Integer similarityGroupCount;

    @Schema(description = "平均相似度")
    private BigDecimal avgSimilarity;

    @Schema(description = "最大相似度")
    private BigDecimal maxSimilarity;

    @Schema(description = "检测阈值")
    private BigDecimal threshold;

    @Schema(description = "相似度分布")
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<Integer> similarityDistribution;

    @Schema(description = "程度统计")
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<CloneLevel> degreeStatistics;

    @Schema(description = "检测模式")
    @Trans(type = TransType.DICTIONARY, key = "CHECK_MODE", ref = "checkModeName")
    private Integer checkMode;

    @Schema(description = "检测模式名称")
    @TableField(exist = false)
    private String checkModeName;
}

File: similarity-service/internal/database/model/task_reports.go
package model

import (
	"time"
)

// 报告库表
type TaskReports struct {
	ID                     string     `gorm:"column:id;primaryKey;type:varchar(32)"`
	ReportType             int        `gorm:"column:report_type;default:0"`
	TaskID                 string     `gorm:"column:task_id;type:varchar(32)"`
	SetID                  string     `gorm:"column:set_id;type:varchar(32)"`
	IsSet                  bool       `gorm:"column:is_set;default:false"`
	ProblemID              string     `gorm:"column:problem_id;type:varchar(32)"`
	SampleCount            int        `gorm:"column:sample_count;default:0"`
	SimilarityGroupCount   int        `gorm:"column:similarity_group_count;default:0"`
	AvgSimilarity          float64    `gorm:"column:avg_similarity;type:decimal(10,2);default:0.00"`
	MaxSimilarity          float64    `gorm:"column:max_similarity;type:decimal(10,2);default:0.00"`
	Threshold              float64    `gorm:"column:threshold;type:decimal(10,2);default:0.50"`
	SimilarityDistribution string     `gorm:"column:similarity_distribution;type:json"`
	DegreeStatistics       string     `gorm:"column:degree_statistics;type:json"`
	CheckMode              int        `gorm:"column:check_mode;default:1"`
	Deleted                bool       `gorm:"column:deleted;default:false"`
	CreateTime             *time.Time `gorm:"column:create_time"`
	CreateUser             string     `gorm:"column:create_user;type:varchar(32)"`
	UpdateTime             *time.Time `gorm:"column:update_time"`
	UpdateUser             string     `gorm:"column:update_user;type:varchar(32)"`
}

func (TaskReports) TableName() string {
	return "task_reports"
}
